pandas/core/indexing.py

```diff
@@ -903,7 +903,7 @@ def _getitem_lowerdim(self, tup):
 
         # we maybe be using a tuple to represent multiple dimensions here
         ax0 = self.obj._get_axis(0)
-        if isinstance(ax0, MultiIndex):
+        if isinstance(ax0, MultiIndex) and self.name != 'iloc':
```

Comment:
add a comment here (or can update the previous one), with the issue number and a 1-line about what is happening here